grub.d/README-internal.md

```diff
@@ -0,0 +1,28 @@
+`grub.d` integration
```

Comment:
@kacf 
I'm finding it hard to follow the thread of "Two styles".
I assume you're naming them `grub-mender-grubenv` and `grub.d`. 
Would appreciate a higher level introduction on why they are here and what they do. Here is an attempt:

> During boot time, grub needs the scripts in this directory to exist in a certain location on the boot partition.
> This process of script installation also involves modifying the scripts (concatenating/renaming). 
> 
> We have two mechanisms for this:
> 
> * `grub-mender-grubenv`
>     * how we historically did it
>     * no external dependencies on grub tools
>     * breaks the bootlader if groob tools are used in paralel
> * `grub.d` - 
>     * standard way used used in Debian
>     * uses `update-grub` and `grub-mkconfig`


Is this correctly interpreted?